x-pack/plugins/apm/public/components/app/TransactionDetails/WaterfallWithSummmary/WaterfallContainer/Waterfall/index.tsx

```diff
@@ -66,72 +68,91 @@ interface Props {
   location: Location;
   exceedsMax: boolean;
 }
-
 export function Waterfall({
   waterfall,
   exceedsMax,
   waterfallItemId,
   location,
 }: Props) {
+  const [isAccordionOpen, setIsAccordionOpen] = useState(true);
   const itemContainerHeight = 58; // TODO: This is a nasty way to calculate the height of the svg element. A better approach should be found
   const waterfallHeight = itemContainerHeight * waterfall.items.length;
 
   const { serviceColors, duration } = waterfall;
 
-  const agentMarks = getAgentMarks(waterfall.entryTransaction);
+  const agentMarks = getAgentMarks(waterfall.entryWaterfallTransaction?.doc);
   const errorMarks = getErrorMarks(waterfall.errorItems, serviceColors);
 
-  function renderWaterfallItem(item: IWaterfallItem) {
-    const errorCount =
-      item.docType === 'transaction'
-        ? waterfall.errorsPerTransaction[item.doc.transaction.id]
-        : 0;
-
+  function renderItems(
+    childrenByParentId: Record<string | number, IWaterfallItem[]>
+  ) {
+    const { entryWaterfallTransaction } = waterfall;
+    if (!entryWaterfallTransaction) {
+      return null;
+    }
     return (
-      <WaterfallItem
-        key={item.id}
+      <AccordionWaterfall
+        // used to recreate the entire tree when `isAccordionOpen` changes, collapsing or expanding all elements.
+        key={`accordion_state_${isAccordionOpen}`}
```

Comment:
Why do we need to recreate the entire tree when this happens?

Reply:
Based on this explanation, https://github.com/elastic/kibana/pull/75259#discussion_r476344748.

To avoid making the `isOpen` prop stateful in the component, once the user pushes the button I recreate the tree changing the default state of the accordion component, so all items will open or close based on the prop received.

Reply:
Gotcha 👍